### PR TITLE
Implemented Webauthn support

### PR DESCRIPTION
**Pull request by LucFauvel**
Added support for webauth to allow the use of Slauth for RPs

### CODE REVIEW

File: src/webauthn/server/mod.rs [hunk 1]
```diff
@@ -0,0 +1,410 @@
...
+    pub credential: PublicKeyCredential,
+    pub credential_pub: CredentialPublicKey,
+    pub context: PublicKeyCredentialRequestOptions,
+    origin: String,
+    user_handle: String,
+    sign_count: u32,
+}
+
+impl CredentialRequestVerifier {
+    pub fn new(credential: PublicKeyCredential, credential_pub: CredentialPublicKey, context: PublicKeyCredentialRequestOptions, origin: &str, user_handle: String, sign_count: u32) -> Self {
```
Comment (richerarc) [inline src/webauthn/server/mod.rs]: `user_handle` shoud be of type `&str` for consistency with other parameters

File: src/webauthn/server/mod.rs [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+        Ok(auth_data.sign_count)
+    }
+}
+
+fn get_ring_alg_from_cose(id: i64, curve: i64) -> Result<&'static dyn VerificationAlgorithm, Error> {
+    if id == WEBAUTHN_COSE_ALGORITHM_IDENTIFIER_EC2 {
+        match curve {
+            ECDSA_CURVE_P256=> Ok(&signature::ECDSA_P256_SHA256_ASN1),
+            ECDSA_CURVE_P384=> Ok(&signature::ECDSA_P384_SHA384_ASN1),
+            _ => Err(Error::Sign("Unsupported algorithm".to_string())),
```
Comment (richerarc) [inline src/webauthn/server/mod.rs]: Should we support `ECDSA_CURVE_P521`?

Comment (LucFauvel) [reply src/webauthn/server/mod.rs]: Sadly, that curve is not supported in ring.

**Comment by richerarc** [comment]
The fuzzer is still missing before merging this PR

**Pull request merged**